git-ftp

```diff
@@ -819,11 +819,10 @@ submodule_catchup() {
 
 set_submodule_args() {
 	args=(--user "$REMOTE_USER")
-	if [ "$REMOTE_PROTOCOL" = "sftp" ] && [ -n "$CURL_PRIVATE_KEY" ] && [ -n "$CURL_PUBLIC_KEY" ]; then
+	[ -n "$REMOTE_PASSWD" ] && args+=(--passwd "$REMOTE_PASSWD")
+	if [ "$REMOTE_PROTOCOL" = "sftp" ] && [ -n "$CURL_PRIVATE_KEY" ]; then
 		args+=(--key "$CURL_PRIVATE_KEY")
-		args+=(--pubkey "$CURL_PUBLIC_KEY")
-	else
-		args+=(--passwd "$REMOTE_PASSWD")
+		[ -n "$CURL_PUBLIC_KEY" ] && args+=(--pubkey "$CURL_PUBLIC_KEY")
```

Comment:
Does this have to be nested? Would there be any downside to writing the whole method like this?
```sh
	args=(--user "$REMOTE_USER")
	[ -n "$REMOTE_PASSWD" ] && args+=(--passwd "$REMOTE_PASSWD")
	[ -n "$CURL_PRIVATE_KEY" ] && args+=(--key "$CURL_PRIVATE_KEY")
	[ -n "$CURL_PUBLIC_KEY" ] && args+=(--pubkey "$CURL_PUBLIC_KEY")
```

It's much easier to read. It's also independent of the usage logic which means that it won't need changing as in much in the future. What do you think?

Reply:
My thought was that it only makes sense to set these parameters (`key` and `pub key`) if sftp is used. As far as I know they are ignored by curl if any other protocol is used. And in my thoughts it was only necessary to set the `pubkey` if there is a (private)`key`.

But yes, you are right. It does not  have to be like this. Since curl is working fine if these are set (even if they are “useless” in a case) we can set set them if they are set in the parent command.